apple/Sources/FerrostarMapLibreUI/Views/DynamicallyOrientingNavigationView.swift

```diff
@@ -69,11 +76,14 @@ public struct DynamicallyOrientingNavigationView: View, CustomizableNavigatingIn
                     camera: $camera,
                     navigationState: navigationState,
                     onStyleLoaded: { _ in
-                        camera = navigationCamera
-                    }
-                ) {
-                    userLayers
-                }
+                        if navigationState?.isNavigating == true {
```

Comment:
Will this be called when the user calls `ferrostarCore.startNavigation(route: route)`?

Reply:
Nope - unless start navigation is called before the style completes. But this call here should only be seen as a fallback, something else should set the camera to navigation camera, like startNavigation.

Reply:
Idea for this concept generally. Could we extend the NavigationMapView with navigations state event view modifier(s) (e.g. `func onNavigationStarted()`). Allowing the developer to apply modifications to the map view's configuration when certain navigationState events occur? This might be a bit overkill, but seems like it would first allow us to do default behaviors like setting the navigationCamera one time when navigation starts as well as let the developer modify the view's behavior when certain nav event occur? It may also be quite scalable as an approach.

Reply:
We *might* not actually need this if we go with your suggestion (which I'll elaborate on in my review comments), since the modifiers get applied any time state changes ;)

Reply:
While we might not need it, it might still be convenient? one can already achieve this by a `.onChange(ferrostar.state)` but this would be easier to use and understand for people starting with the framework?

Reply:
Yeah, fair point ;) I'm open to draft implementations from either of you if you think it's helpful. I originally was super enthusiastic when @Archdoog described it on our call but then realized it might not be a hard requirement with the other proposed changes.